sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -2079,8 +2082,8 @@ def collect_pipeline_signatures(root_dict: dict,
                 string = '#    ' + signature['name'] + ': ' + signature[
                     'parameterType']
                 if 'defaultValue' in signature:
-                    string += ' [Default: ' + str(
-                        signature['defaultValue']) + ']'
+                    string += ' [Default: ' + escape_linebreak_characters(
+                        str(signature['defaultValue'])) + ']'
```

Comment:
```suggestion
                       signature['defaultValue']) + ']'
```

Reply:
Explicitly type casting is needed here, since the default value will not be implicitly casted when string concatenation is performed via addition.

Since converting all uses of addition to f-strings in this function would distract from the intent of the PR, I'll omit this change.